packages/wasm-parser/src/index.js

```diff
@@ -1,14 +1,51 @@
 // @flow
-
+import { traverse } from "@webassemblyjs/ast";
 import * as decoder from "./decoder";
 
 const defaultDecoderOpts = {
   dump: false,
   ignoreCodeSection: false,
-  ignoreDataSection: false
+  ignoreDataSection: false,
+  ignoreCustomNameSection: false
 };
 
+function restoreNames(ast) {
```

Comment:
Instead of doing a separate pass couldn't we do this in the decoder directly?

Reply:
I don't think this is possible. The name section appears at the very end of the module, so the only way to apply the naming is via a second pass.

Reply:
Ok that make sense.

According to the spec a custom section can be after each non-custom section, so depending on how the binary was constructed It could happens, but we can leave as is since it will work in any case.